version.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+if [ -n "$PROJECT_VERSION" ]; then
+    # use env variable PROJECT_VERSION if defined
+    echo $PROJECT_VERSION
+elif ! [ -x "$(command -v git)" ] || ! [ -d .git ]; then
+	# git is not installed, or not in a git repo
+    echo '0-unknown-version'
+else
+	# fetch all tags silently
+	git fetch --tags --quiet > /dev/null 2> /dev/null
+	# get human readable commit id from git
+	git describe --tags --always --dirty | sed 's/^v//'
```

Comment:
If git describe always goes back to a tag, does it mean that portal dependencies can only be brought in from a tagged commit that is pushed to the cbioportal github repo? If a single PR makes changes in both the service and web layer (like changing the service interface), will the "new" web layer be able to bring in the "new" service layer even if a tag has not been created (in a pr to rc, for instance)? Maybe the extra info after the tag is enough to provide the additional path since the tag (the "dirty" part I guess?) 

% git describe --tags --always --dirty

v1.18.0-120-g05c7ad6-dirty

Reply:
@sheridancbio Portal dependencies can be brought in locally or remotely. It doesn't need to be tagged/committed in the github repo. Maven continues to function in the same way, so if u compile something locally the artifacts will end up in your `~/.m2` folder and you can include them in another project as a dependency.

Reply:
sounds good. I was mainly thinking about dependencies between the modules within cbioportal, but packaging the cbioportal code into other codebases is also needed.